packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +266,374 @@ export default class VrackMoveDialogCtrl {
+      .then((cloudProject) => {
+        this.data.eligibleServices.cloudProject = cloudProject;
+      });
+  }
+
+  // Get service info for each dedicated cloud retrieved from API
+  updateDedicatedCloudServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.dedicatedCloud.map((serviceId) => {
+          return this.vrackService
+            .getDedicatedCloud(serviceId)
+            .then((response) => {
+              const dedicatedCloud = VrackMoveDialogCtrl.getDedicatedCloudNiceName(
+                response.data,
+              );
+              return dedicatedCloud;
+            });
+        }),
+      )
+      .then((dedicatedCloud) => {
+        this.data.eligibleServices.dedicatedCloud = dedicatedCloud;
+
+        const dedicated = this.data.eligibleServices.dedicatedCloud.filter(
+          (service) =>
+            service.productReference === DEDICATED_CLOUD_PRODUCT.epcc,
+        );
+        const managedBareMetal = this.data.eligibleServices.dedicatedCloud.filter(
+          (service) => service.productReference === DEDICATED_CLOUD_PRODUCT.mbm,
+        );
+        this.data.eligibleServices.dedicatedCloud = dedicated;
+        this.data.eligibleServices.managedBareMetal = managedBareMetal;
+      });
+  }
+
+  // Get service info for each dedicated server retrieved from API
+  updateDedicatedServerServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.dedicatedServer.map((serviceId) => {
+          return this.vrackService
+            .getDedicatedServer(serviceId)
+            .then((response) => {
+              const dedicated = VrackMoveDialogCtrl.getDedicatedServerNiceName(
+                response.data,
+              );
+              return dedicated;
+            });
+        }),
+      )
+      .then((dedicatedServer) => {
+        this.data.eligibleServices.dedicatedServer = dedicatedServer;
+      });
+  }
+
+  updateDedicatedServerInterfaceServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.dedicatedServerInterface.map((service) => {
+          return this.vrackService
+            .getDedicatedServerInterface(service)
+            .then((response) => {
+              let dedicatedServerInterfaceUpdate = service;
+              dedicatedServerInterfaceUpdate.dedicatedServer = response.data;
+              dedicatedServerInterfaceUpdate = VrackMoveDialogCtrl.getDedicatedServerInterfaceNiceName(
+                dedicatedServerInterfaceUpdate,
+              );
+              return dedicatedServerInterfaceUpdate;
+            });
+        }),
+      )
+      .then((dedicatedServerInterface) => {
+        this.data.eligibleServices.dedicatedServerInterface = dedicatedServerInterface;
 
         // We need to append dedicatedServerInterfaces list to dedicatedServers list.
         if (
-          has(allServices, 'dedicatedServerInterface') &&
-          allServices.dedicatedServerInterface.length > 0
+          has(
```

Comment:
we try to avoid lodash or underscore usage (even if it is true that our legacy is using it)

Reply:
I just move some code lines.